src/plugins/visualizations/public/actions/edit_in_lens_action.tsx

```diff
@@ -50,11 +63,24 @@ const MenuItem: React.FC = () => {
   );
 };
 
-const isVisualizeEmbeddable = (embeddable: IEmbeddable): embeddable is VisualizeEmbeddable => {
-  return 'getVis' in embeddable;
-};
+type EditInLensActionApi = HasUniqueId &
+  HasVisualizeConfig &
+  PublishesPanelTitle &
+  PublishesPanelDescription &
+  Partial<PublishesLocalUnifiedSearch> &
+  CanAccessViewMode &
+  Partial<HasExpressionVariables>;
 
-export class EditInLensAction implements Action<EditInLensContext> {
+const compatibilityCheck = (api: EmbeddableApiContext['embeddable']): api is EditInLensActionApi =>
```

Comment:
Why are we adding so many checks? This seems a bit complicated to me. Can you elaborate? Maybe we can find a better and simpler way for the consumers to do this? 🤔

Reply:
The amount of detail in the type guards is a trade off between verbosity and runtime safety. If you wanted to, it would be completely possible to use a type guard that checks only for the presence of getVis and assumes that the other required keys are present and are of the right types. That is roughly how it worked before. This is less code, but is more likely to result in runtime errors, as there could theoretically be an API that has getVis, but doesn't have an ID for instance.

IMO balancing this trade off is up to the consumer. The Presentation team tends to lean towards verbosity, because each type guard describes exactly what it checks for and corresponds to an interface used in the Action's required API. I.e. if PublishesPanelTitle is required, the type guard should run apiPublishesPanelTitle.

Again though, this is not strictly required.

Reply:
The new embeddable system is **very** declarative - i.e. we are being **very** explicit about what it takes for an action to be compatible with a given API/embeddable. @ThomThomson might be able to help give more context here 🙇

Reply:
ok it still feels very complicated to me 😄 Should everytime the consumer know all these guards? Canw e create a function or 2 which group these things somehow? So the consumer is using only this function and doesnt need to know all these guards

Reply:
TBH most actions only require a couple of these - usually 3 or 4. I'd say this is a special case.

Creating a higher-order `apiIsEmbeddable` type guard that checks for all the common functionalities of Embeddables at once is possible yes.

**Pros:** 
- Less code
- Easier to check all common interfaces at once (might encourage developers to be safer at runtime)

**Cons:**
- Could result in checking for keys that are not used. (I.e. if the API only uses `id`, should we be checking the presence for all 10 of the other keys?)
- Might only be used once or twice because most actions don't need so many checks.

Reply:
Well I think we need to have some utilities here:

- Which are the ones we have for sure everywhere? --> one function. These might be apiHasUniqueId(api), apiPublishesPanelTitle, apiPublishesPanelDescription, apiPublishesPartialLocalUnifiedSearch, apiCanAccessViewMode, getInheritedViewMode
- Let the unique ones as they are

I am honestly telling you that as a developer I find this complicated check difficult to read and next time I want to write an action I will be super confused on what to do 🤷‍♀️ 

So I would:

- create one utility in presentation-utils which gathers the guards that make sense for the embeddabled
- create one utility in visualizations plugin which gathers the guards for legayc visualizations
- combine them and use them to the action

Reply:
This approach sounds good to me!

Reply:
Can do 🖖

Reply:
@stratoula Worked with @ThomThomson offline to simplify things a bit in 11b86d41dc3634934a3888b8c1ebc0cfbb0fbce5.

Can you take another look?